### ISSUE TITLE
`no-untyped-call` false positive when calling lambda with no input parameters

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#14473]
```py
from typing import Callable
(lambda: 1)()  # no error
a = lambda: 1
b: Callable[[], int] = lambda: 1
reveal_type(a)  # Revealed type is "def () -> builtins.int"
reveal_type(b)  # Revealed type is "def () -> builtins.int"
a()  # error: Call to untyped function (unknown) in typed context  [no-untyped-call]
b()  # no error
```
[playground](https://mypy-play.net/?mypy=latest&python=3.11&flags=show-error-context%2Cshow-column-numbers%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cno-implicit-optional%2Cno-implicit-reexport%2Cno-strict-optional%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=5c4e4c1bfb99c539100720712b07eeda)

kinda related: #5069
